corehq/apps/reports/standard/cases/utils.py

```diff
@@ -170,9 +170,9 @@ def _get_location_accessible_ids(request):
 
 def query_location_restricted_cases(query, request):
     accessible_ids = _get_location_accessible_ids(request)
-    return case_es.owner(accessible_ids)
+    return query.filter(case_es.owner(accessible_ids))
```

Comment:
I couldn't tell if `case_es.owner(accessible_ids)` returned a dict, but it was curious to me that these functions did not use the `query` argument at all.

Reply:
that's where `ipdb` becomes your friend :smiley_cat:

Reply:
I think what happened was I was expecting these functions to return a filter, since that's typically how these query abstractions work, but they actually return a modified `query` object.  The code confusingly used an OR filter over a single argument, which made it look like the OR joined the query and the filter, instead of joining variable number of filter argument.

```python
query.OR(case_es.owner(accessible_ids))
# is equivalent to
query.filter(filters.OR(case_es.owner(accessible_ids)))
# which is logically equivalent to
query.filter(case_es.owner(accessible_ids))
# which I realize now can also be written more simply as
query.owner(accessible_ids)
```

Reply:
> that's where `ipdb` becomes your friend smiley_cat

I didn't know of an easy way to reproduce the issue from the ticket in `ipdb`.